deployment/terraform/create.go

```diff
@@ -279,6 +279,26 @@ func (t *Terraform) setupAppServer(extAgent *ssh.ExtAgent, ip, serviceFile strin
 		{srcData: strings.TrimSpace(serviceFile), dstPath: "/lib/systemd/system/mattermost.service"},
 		{srcData: strings.TrimPrefix(limitsConfig, "\n"), dstPath: "/etc/security/limits.conf"},
 	}
+
+	// Specify a hosts file when the ServerHostname is set, so that it points to
+	// either the proxy IP or, if there's no proxy, to localhost.
+	if t.config.ServerHostname != "" {
+		output, err := t.Output()
+		if err != nil {
+			return err
+		}
+
+		ip := "127.0.0.1"
+		if output.HasProxy() {
+			ip = output.Proxy.PrivateIP
+		}
```

Comment:
Of course, that's what I meant to write here :man_facepalming:

Reply:
Perhaps I am missing something, but how is this supporting the single server setup? In that case it will be `127.0.0.1 ltserver`. But the actual IP will be something else right?

Reply:
The `ltserver` domain name will point to the same machine where it's defined, so when the server running there requests something from `ltserver`, it will request it actually to itself. We could use the private or public IP as well if that's less confusing.

There's a bug with the new changes, though, since in the single server setup, the server's `SiteURL` should contain the port as well, as in http://ltserver:8065, which is where the server is actually listening.

Reply:
We could use even the public DNS, which is what we show when running `deployment info`.

Reply:
Scratch that, /etc/hosts entries can only point to IPs :)

Reply:
I've also commented the whole configuration of the server's `SiteURL` to explain all cases: https://github.com/mattermost/mattermost-load-test-ng/pull/645/commits/081e460e9c957a9de6ec92d6c9648194d08b6377

Reply:
Ugh, the single-server setup is driving me crazy, since in that case:
- We need to add the port to the server's `SiteURL`, as in `http://ec2-18-212-134-134.compute-1.amazonaws.com:8065`, since the single node is listening on 8065 (not in 80, as the proxy does).
- If we add a user-provided `SiteURL` to override the server's, as `ltserver`, then we need to add the port as well to the server's `SiteURL`, as `ltserver:8065`, since the entry in `/etc/hosts` needs to point to a plain IP, it cannot point to a specific port.
- But of course, the database dump's permalinks will contain the `ltserver` hostname with no port (so we can cover the more interesting cases with multiple nodes), so the server will not recognize those as permalinks (it needs the whole `SiteURL` to show up in the link to assume it's a permalink).

We can:
- Forget about the single-node setup and focus only on the multiple-nodes setup.
- Spin up a proxy unconditionally, so that even for single-server setups we have a proxy to which we can point the SiteURL to.
- Have another version of the database with the 8065 port added in the permalinks.

Thoughts?

Reply:
1 sounds fine to me.

Reply:
> The ltserver domain name will point to the same machine where it's defined, so when the server running in that machine requests something from ltserver, it will request it actually to itself. We could use the private or public IP as well if that's less confusing.

But the host file entry is on the load test agents right? So if `ltserver` resolves to 127.0.0.1 on the agent, then how would it work? That was my question.

Reply:
> But the host file entry is on the load test agents right?

No, the host file entry is on the server. The agents do not need to follow those links (they do parse them, but only matching the `/pl/` part, and get the post ID,  then forgetting about the link itself). The server does not follow the links either, but it needs to see the exact site URL in the link to assume it's a permalink (the `looksLikeAPermalink` function). So we need to reconcile those two things:
- Server's `SiteURL` needs to be a valid URL that points to either the proxy or to the machine itself in case of a single node. Here's where the server's /etc/hosts come into play.
- Permalinks need to have the exact `SiteURL` as their domain name, so that the server recognizes them as permalinks. This is where we need to tune the new config setting `SiteURL` (whose name I'll change because it's impossible to talk about both server's `SiteURL` and loadtest config's `SiteURL` without becoming crazy)

Reply:
Aha .. so that was my understanding. I thought the agents were resolving the links. Got it. So let's go with:

> Forget about the single-node setup and focus only on the multiple-nodes setup.

Reply:
> 1 sounds fine to me.

Actually, with the current code, we can use both single-node and multiple-node setups with one single database. The database needs to have the permalinks with the domain name as `ltserver:8065`, and the server will recognize those permalinks both when the siteurl is ltserver and when the siteurl is ltserver:8065, since the looksLikeAPermalink function matches `${SiteURL}.*/pl/`, so the additional port is matched by the generic `.*`

So yeah, I'm happy with what we have now.